tests/frontend/messages_access_test.js

```diff
@@ -0,0 +1,109 @@
+ */
+
+/* global expect, describe, it */
+
+import {gettext, ngettext, ngettextCustom} from "../../warehouse/static/js/warehouse/utils/messages-access";
+
+
+describe("messages access util", () => {
+
+  describe("gettext with defaults", () => {
+    it("uses default singular when no translation is available", async () => {
+      const singular = "My default message.";
+      const result = gettext(singular);
+      expect(result).toEqual(singular);
+    });
+    it("inserts placeholders into the default singular", async () => {
+      const singular = "My default message: %1";
+      const extras = ["more message here"];
+      const result = gettext(singular, ...extras);
+      expect(result).toEqual("My default message: more message here");
+    });
+  });
+
+  describe("ngettext with defaults", () => {
+    it("uses default singular when no translation is available", async () => {
+      const singular = "My default message.";
+      const plural = "My default messages.";
+      const num = 1;
+      const result = ngettext(singular, plural, num);
+      expect(result).toEqual(singular);
+    });
+    it("inserts placeholders into the default singular", async () => {
+      const singular = "My %2 default %1 message.";
+      const plural = "My default messages.";
+      const num = 1;
+      const extras = ["more message here", "something else"];
+      const result = ngettext(singular, plural, num, ...extras);
+      expect(result).toEqual("My something else default more message here message.");
+    });
+    it("uses default plural when no translation is available", async () => {
+      const singular = "My default message.";
+      const plural = "My default messages.";
+      const num = 2;
+      const result = ngettext(singular, plural, num);
+      expect(result).toEqual(plural);
+    });
+    it("inserts placeholders into the default plural", async () => {
+      const singular = "My %2 default %1 message.";
+      const plural = "My default plural messages %1 %2.";
+      const num = 2;
+      const extras = ["more message here", "something else"];
+      const result = ngettext(singular, plural, num, ...extras);
+      expect(result).toEqual("My default plural messages more message here something else.");
+    });
+  });
+
+  describe("with translation data", () => {
+    const data = {
+      "": {"language": "fr", "plural-forms": "nplurals=2; plural=n > 1;"},
+      "My default message.": "My translated message.",
+      "My %2 message with placeholders %1.": "My translated %1 message with placeholders %2",
+      "My message with plurals": ["My translated message 1.", "My translated messages 2."],
+      "My message with plurals %1 again": ["My translated message 1 %1.", "My translated message 2 %1"],
+    };
+    const pluralForms = function (n) {
+      let nplurals, plural;
+      nplurals = 2; plural = n > 1;
+      return {total: nplurals, index: ((nplurals > 1 && plural === true) ? 1 : (plural ? plural : 0))};
+    };
```

Comment:
question: this looks a lot like a parametrized test, is that the case here? If so, would it make more sense to use [jest's `.each`](https://jestjs.io/docs/api#each) to express the individual tests to reduce the amount of logic baked in to the test case?